Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -699,6 +700,38 @@ void RenderGrid::computeIntrinsicLogicalWidths(LayoutUnit& minLogicalWidth, Layo
 
     computeTrackSizesForIndefiniteSize(algorithm, GridTrackSizingDirection::ForColumns, gridLayoutState, &minLogicalWidth, &maxLogicalWidth);
 
+    if (isMasonry(GridTrackSizingDirection::ForColumns)) {
+        // The track sizing algorithm will only be run once in this case, since track sizing will not run in the masonry direction.
+        computeTrackSizesForIndefiniteSize(algorithm, GridTrackSizingDirection::ForRows, gridLayoutState, &minLogicalWidth, &maxLogicalWidth);
+
+        auto gridAxisTracksCountBeforeAutoPlacement = currentGrid().numTracks(GridTrackSizingDirection::ForRows);
+        LayoutUnit availableLogicalWidth;
+
+        for (auto* gridItem = grid.orderIterator().first(); gridItem; gridItem = grid.orderIterator().next()) {
```

Comment:
I think this whole thing deserves a comment explaining our rationale behind why we are overriding the logical width of the grid items because it is not obvious imo